synapse/storage/databases/main/events_worker.py

```diff
@@ -1299,7 +1303,7 @@ async def _fetch_event_ids_and_get_outstanding_redactions(
                 # invites, so just accept it for all membership events.
                 #
                 if d["type"] != EventTypes.Member:
-                    raise Exception(
+                    raise InvalidEventError(
```

Comment:
Not the goal of this PR, but in passing: Is it possible that there's no `type` field in `d`?